rfcs/0000-observability-api.md

```diff
@@ -0,0 +1,223 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The observability api, will allow a user to choose different parts of the offered Kuadrant observability. This provides more freedom to the user rather then the all or nothing current approach for certain aspects or the manual entry in multiple locations approach. With the new way the user will only have to fill in their configuration in one location.
+
+### Potential configurations
+
+The different aspects a user might want to modify could be the following:
+
+| Observability piece   | Kuadrant component                                        | Options                                                                                                                       |
+|-----------------------|-----------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------|
+| Logging               | Kuadrant operator, Authorino operator, Limitador Operator | **component**  **logLevel**:  **logMode**:                                                           |
+| Tracing               | Authorino operator, Limitador operator                    | **component** , **endpoint** , **tags**  , **insecure**, **strageyRules**  |
+| Metrics               | Kuadrant operator, Authorino, Limitador, DNS Operator     | **component** , **enableService** , **port** , **deep**                                            |
+| Alerts *              | Kuadrant operator, Authorino, Limitador                   | **namespace** , **component**, **enable**                                                                 |
+| Dashboards *          | Kuadrant operator, Authorino, Limitador                   | **namespace** , **component**, **enable**                                                                 |
+| Other 3rd Party *     | e.g Kiali                                                 | **enable** bool                                                                                                               |
+
+###### **Note**: Observability pieces with a * denotes these are post v1 milestone
+
+### Example CR with everything
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: observability
+spec:
+      logging:
+        component:
+          authorino:
+            logLevel: debug
+          limitador: 
+            logMode: debug
+          authorino: 
+            logMode: development
+          limitador: 
+            logMode: development  
+    
+    tracing:
+      component:  
+        limitador:
+          endpoint: rpc://tempo.tempo.svc.cluster.local:4317
+          insecure: true
+          tags: tag1, tag2
+          strategyRules: 
```

Comment:
incorporating potential other RFC https://github.com/Kuadrant/architecture/pull/96

Reply:
Maybe a note to say this section will depend on work in that RFC?

Reply:
Sure ill add that in